src/Test/Utilities/Shared/Diagnostics/DiagnosticDescription.cs

```diff
@@ -463,6 +439,11 @@ public static string GetAssertText(DiagnosticDescription[] expected, IEnumerable
                 }
 
                 var description = new DiagnosticDescription(d, errorCodeOnly: false, showPosition: true);
+                var idx = Array.IndexOf(expected, description);
+                if (idx != -1)
+                {
+                    description = expected[idx];
```

Comment:
>description = expected[idx]; [](start = 20, length = 28)

Based on off-line conversation, this is going to affect content of the "Actual" portion of the output as well. If baseline doesn't have location for a particular diagnostics, the "Actual" won't show location any more. I think we would prefer to always have locations in the "Actual" portion. Ideally, all baselines should have locations and having them in "Actual" would make it easier to get to that state for existing tests over time.